internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/util/FileUtil.java

```diff
@@ -0,0 +1,314 @@
+            File newFile = new File(source);
+            File[] fileList = newFile.listFiles();
+            if (fileList != null) {
+                for (File file : fileList) {
+                    ZipArchiveEntry entry = new ZipArchiveEntry(file.getPath());
+                    archive.putArchiveEntry(entry);
+                    try (BufferedInputStream input = new BufferedInputStream(new FileInputStream(file))) {
+                        IOUtils.copy(input, archive);
+                    } finally {
+                        archive.closeArchiveEntry();
+                    }
+                }
+            } else {
+                log.warn("File list that needs to be archived cannot be null.");
+            }
+            archive.finish();
+        }
+    }
+
+    /**
+     * Download the product pack from the FTP location.
+     *
+     * @param productPathInFtp path to download pack in the FTP location.
+     * @param productName      file name to download.
+     * @throws ScannerException
+     */
+    public static void downloadProduct(String productPathInFtp, String productName, File file) throws
+            IOException, SftpException, JSchException {
+        ChannelSftp sftp = goToFtpProductLocation(productPathInFtp);
+
+        downloadFromFtp(sftp.get(productName), file);
+        sftp.disconnect();
+    }
+
+    /**
+     * Go to the FTP location where the product pack is.
+     *
+     * @param productPathInFtp path to the product pack
+     * @return SFTP channel to access the FTP location
+     * @throws JSchException
+     * @throws SftpException
+     */
+    private static ChannelSftp goToFtpProductLocation(String productPathInFtp) throws JSchException, SftpException {
+        JSch jsch = new JSch();
+        Session session;
+        Channel channel;
+        ChannelSftp sftp;
+
+        // Get the FTP configuration to build the connection with FTP server.
+        String ftpHost = YAMLConfigurationReader.getInstance().getConfigProperty(ScannerConstants.FTP_HOST);
+        String ftpUsername = YAMLConfigurationReader.getInstance().getConfigProperty(ScannerConstants.FTP_USERNAME);
+        char[] ftpPassword = (YAMLConfigurationReader.getInstance().getConfigProperty(ScannerConstants.FTP_PASSWORD))
+                .toCharArray();
+        int ftpPort = Integer.parseInt(YAMLConfigurationReader.getInstance().getConfigProperty(ScannerConstants
+                .FTP_PORT));
+
+        session = jsch.getSession(ftpUsername, ftpHost, ftpPort);
+        session.setConfig("StrictHostKeyChecking", "no");
+        session.setPassword(String.valueOf(ftpPassword));
+        session.connect();
+
+        channel = session.openChannel(ScannerConstants.SFTP);
+        sftp = (ChannelSftp) channel;
+        sftp.connect();
+        sftp.cd(productPathInFtp);
+
+        cleanFtpPassword(ftpPassword);
+
+        return sftp;
+    }
+
+    /**
+     * Download the product from FTP to local location.
```

Comment:
This is a util method. Hence, let's not mention about `product` here. Instead make the doc comments generic.